lib/karafka/connection/client.rb

```diff
@@ -70,6 +70,15 @@ def batch_poll
         @buffer
       end
 
+      # Polls Kafka with a set timeout as a keep alive
+      #
+      # @param timeout in milliseconds to wait for poll
+      # @note This is intended only as a keep alive for the consumer - the partition|topic
+      #   should be paused before using this to not repeatedly read messages
+      def keep_alive_poll(timeout:)
+        poll(timeout)
```

Comment:
Ah yes ok I missed this.

Assuming your testing proves this is correct (And I think it will) - I can see a few ways around this limitation.

1. Like you say we could look at the `One Consumer Per Thread` model
> One Consumer Per Thread
> A simple option is to give each thread its own consumer instance. Here are the pros and cons of this approach:
> - PRO: It is the easiest to implement
> - PRO: It is often the fastest as no inter-thread co-ordination is needed
> - PRO: It makes in-order processing on a per-partition basis very easy to implement (each thread just processes messages in the order it receives them).
> - CON: More consumers means more TCP connections to the cluster (one per thread). In general Kafka handles connections very efficiently so this is generally a small cost.
> - CON: Multiple consumers means more requests being sent to the server and slightly less batching of data which can cause some drop in I/O throughput.
> - CON: The number of total threads across all processes will be limited by the total number of partitions.
  - Thats actually listed here: [KafkaConsumer Documentation: Multi-threaded Processing](https://kafka.apache.org/31/javadoc/org/apache/kafka/clients/consumer/KafkaConsumer.html#multithreaded) as one of the options for multi-threaded processing of records.
  - Definitely simplifies the code underneath it.

2. We could run the whole ActiveJob `consume` in a thread
- Still _pause_ the **topic** + **partition** on starting the message consumption meaning the main thread above that continues to run `batch_poll` in a `fetch_loop` acting as the keep alive for the Kafka consumer and not blocking other topics / partitions while only receiving messages again when the thread below _resumes_ the **topic** + **partition** 

Thoughts on those options?